### PR TITLE
Features/first iteration fixes

### PR DESCRIPTION
**Pull request by other91**
bugs fixed:
- empty CommentCell
- freezes scrolling commentsTableView

### CODE REVIEW

File: ITRStartTask/AppLibrary/ApiService/ApiService.swift [hunk 1]
```diff
@@ -19,54 +19,26 @@ class ApiService: StoriesApiServiceProtocol, CommentsApiServiceProtocol {
...
                          onSuccess: @escaping ([CommentItem]) -> Void,
                          onFailure: @escaping (Error) -> Void) {
-        getFeedItems(ids: ids, onSuccess: onSuccess, onFailure: onFailure)
+        getFeedItems(byIds: ids, onSuccess: onSuccess, onFailure: onFailure)
     }
 
-    private func getStoryIds(type: String,
-                             onSuccess: @escaping ([Int]) -> Void,
-                             onFailure: @escaping (Error) -> Void) {
+    func getStoryAllIds(type: String,
```
Comment (asleptsov-itr) [inline ITRStartTask/AppLibrary/ApiService/ApiService.swift]: getStoryAllIds <<-->> type??

File: ITRStartTask/AppLibrary/ApiService/ApiService.swift [hunk 2]
```diff
@@ -96,59 +68,46 @@ class ApiService: StoriesApiServiceProtocol, CommentsApiServiceProtocol {
         task.resume()
     }
 
-    func getStoryItemUrlsWith (ids: [Int]) throws -> [URL] {
+    func getGeedItemUrls (forIds ids: [Int]) -> [URL] {
```
Comment (asleptsov-itr) [inline ITRStartTask/AppLibrary/ApiService/ApiService.swift]: remove whitespace before (

File: ITRStartTask/AppLibrary/ApiService/ApiService.swift [hunk 3]
```diff
@@ -96,59 +68,46 @@ class ApiService: StoriesApiServiceProtocol, CommentsApiServiceProtocol {
...
-                for operation in operations {
-                    if let error = operation.error {
-                        onFailure(error)
-                        return
-                    }
-
-                    if let storyItem: T = operation.storyItem {
-                        storyItems.append(storyItem)
-                    }
+        let urls = getGeedItemUrls(forIds: ids)
```
Comment (asleptsov-itr) [inline ITRStartTask/AppLibrary/ApiService/ApiService.swift]: getGeedItemUrls --> Feed

File: ITRStartTask/AppLibrary/ApiService/Operations/FeedItemDownloadAsyncOperation.swift [hunk 4]
```diff
@@ -91,7 +97,11 @@ class FeedItemDownloadAsyncOperation<T: Codable>: AsyncOperation {
                 return
             }
 
-            onFailure(FeedItemDownloadAsyncOperationError.not200.error())
+            if data == nil {
+                onFailure(FeedItemDownloadAsyncOperationError.not200.error())
```
Comment (asleptsov-itr) [inline ITRStartTask/AppLibrary/ApiService/Operations/FeedItemDownloadAsyncOperation.swift:101]: "return" missed

File: ITRStartTask/AppLibrary/ApiService/StoriesApiServiceProtocol.swift [hunk 5]
```diff
@@ -6,14 +6,45 @@
...
+    case topStories = "topstories"
+    case newStories = "newstories"
+    case bestStories = "beststories"
+
+    func name() -> String {
+        return self.rawValue
+    }
+}
 
 protocol StoriesApiServiceProtocol {
```
Comment (asleptsov-itr) [inline ITRStartTask/AppLibrary/ApiService/StoriesApiServiceProtocol.swift]: refactor

File: ITRStartTask/StoryDetail/Presenter/StoryPresenter.swift [hunk 6]
```diff
@@ -31,7 +31,7 @@ class StoryPresenter: StoryPresenterProtocol {
             let ids = ids.count > 5 ? Array(ids[0..<5]) : ids
```
Comment (asleptsov-itr) [inline ITRStartTask/StoryDetail/Presenter/StoryPresenter.swift]: Magic numbers should be removed from code...

File: ITRStartTask/StoryDetail/Views/StoryViewController.swift [hunk 7]
```diff
@@ -31,6 +31,10 @@ class StoryViewController: UIViewController {
 
         presenter.show()
     }
+
+    static func xibInstance () -> StoryViewController {
```
Comment (asleptsov-itr) [inline ITRStartTask/StoryDetail/Views/StoryViewController.swift]: move to UIViewcontroller extension

File: ITRStartTask/AppLibrary/ApiService/ApiService.swift (hunk 2 above)
Comment (other91) [reply ITRStartTask/AppLibrary/ApiService/ApiService.swift]: fixed

File: ITRStartTask/AppLibrary/ApiService/ApiService.swift (hunk 3 above)
Comment (other91) [reply ITRStartTask/AppLibrary/ApiService/ApiService.swift]: fixed

File: ITRStartTask/AppLibrary/ApiService/Operations/FeedItemDownloadAsyncOperation.swift (hunk 4 above)
Comment (other91) [reply ITRStartTask/AppLibrary/ApiService/Operations/FeedItemDownloadAsyncOperation.swift:101]: fixed

File: ITRStartTask/AppLibrary/ApiService/ApiService.swift (hunk 1 above)
Comment (i2Homyakov) [reply ITRStartTask/AppLibrary/ApiService/ApiService.swift]: Maybe a category?

Comment (other91) [reply ITRStartTask/AppLibrary/ApiService/ApiService.swift]: fixed

File: ITRStartTask/StoryDetail/Presenter/StoryPresenter.swift (hunk 6 above)
Comment (other91) [reply ITRStartTask/StoryDetail/Presenter/StoryPresenter.swift]: fixed

File: ITRStartTask/StoryDetail/Views/StoryViewController.swift (hunk 7 above)
Comment (other91) [reply ITRStartTask/StoryDetail/Views/StoryViewController.swift]: fixed

File: ITRStartTask/AppLibrary/ApiService/StoriesApiServiceProtocol.swift (hunk 5 above)
Comment (other91) [reply ITRStartTask/AppLibrary/ApiService/StoriesApiServiceProtocol.swift]: StoriesApiServiceProtocol refactored

File: ITRStartTask/AppLibrary/Extensions/UIVIewController(XibInstance).swift [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+//
+
+import UIKit
+
+extension UIViewController {
+    static func xibInstance<T: XibInitializable & UIViewController>() -> T {
+        let nibName = String(describing: self)
+        return T(nibName: nibName, bundle: nil)
+    }
+}
```
Comment (asleptsov-itr) [inline ITRStartTask/AppLibrary/Extensions/UIVIewController(XibInstance).swift:16]: static func xibInstance() -> UIViewController {
        let viewController = self.init(nibName: String(describing: self), bundle: nil)
        return viewController
    }

**Pull request merged**